### PR TITLE
EQL: Update between() case handling and tests

### LINKED ISSUE #56749: EQL: implement case sensitivity for between string function
Following a similar approach in https://github.com/elastic/elasticsearch/pull/56404/, implement case sensitivity based on request parameter for `between` string function.

### PR DESCRIPTION
**Pull request by rw-access**
Closes #56749

* Made the `between()` function configuration aware and remove the deprecated `case_sensitive` argument.
* Updated the `null` handling to pass tests and match this specification
    * if `greedy` is unspecified, then it is set to `false` (unchanged)
    * if any of the input arguments (`source`, `left`, `right`, `greedy`) are `null`, then the function will return null
    * if `left` is not in the string `source` or `right` is not in the string `source`, then `null` is returned
* Updated the handling for empty strings to match the specs/test
    * if `right` is an empty string and `greedy` is true, then the rest of the string after `left` is consumed

### CODE REVIEW

**Comment by costin** [comment]
I'll be on PTO in the upcoming weeks so I removed myself from the reviewers list.
Please wait for both their approval before merging this in.

Thanks,

**Review by matriv — COMMENTED** [review]
Looks good, but maybe add some test that if either of the args is null => null is returned.
And from what I see we need to update the docs to describe that null behaviour.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java
Comment (rw-access) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java:36]: ++. what's our convention here? looks like it was indented twice before, so I unindented the other lines so it's all indented once. I don't know if this comes down to IDE or preference. personally I prefer the single indent between `{` and `}` but can set them all back to two (soft) tabs

Comment (matriv) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java:36]: I don't know, you can check with `./gradlew :x-pack:plugin:eql:spotlessApply`, warning that will change all files and you need to revert, but at least let's keep it consistent: so either fix this line or apply the same indentation to the rest.

Comment (rw-access) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java:36]: thanks for the pointer. spotless choose 4 spaces so I made them all consistent, but backed out the other irrelevant changes to keep the PR smaller

**Comment by matriv** [comment]
> Looks good, but maybe add some test that if either of the args is null => null is returned.
> And from what I see we need to update the docs to describe that null behaviour.

@rw-access Please don't forget ^^ .

**Comment by rw-access** [comment]
@matriv re: null behavior

I was relying on these tests
https://github.com/elastic/elasticsearch/blob/e9b1fa58cf8d510a4b4afb14f66b0d5f9c603ebb/x-pack/plugin/eql/src/test/resources/test_string_functions.toml#L1-L58

but I'll add more tests to https://github.com/elastic/elasticsearch/blob/5c978ac89f6b39de2c118e7a85eb4053ac7edcd6/x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/StringUtilsTests.java#L81

**Comment by matriv** [comment]
Ah, I see, not insisting, just it would be nice to have a couple in Java unit tests too.

**Review by astefan — APPROVED** [review]
LGTM in general. I agree with @matriv about Java unit tests. The more, the better.
Also, it's surprising to me that the case sensitivity for `between` didn't make it for the first (experimental) release of eql.

**Comment by rw-access** [comment]
@elasticmachine update branch

**Pull request closed without merging**
